cocos/core/game.ts

```diff
@@ -368,8 +368,8 @@ export class Game extends EventTarget {
      * @en Run the game frame by frame.
      * @zh 执行一帧游戏循环。
      */
-    public step () {
-        legacyCC.director.mainLoop();
+    public step (now?: number) {
+        legacyCC.director.mainLoop(now);
```

Comment:
Another choose is to pass the parameter as delta time so the implementation becomes:
```ts
public step (deltaTime?: number) {
    // deltaTime is in seconds?
    legacyCC.director.mainLoop(typeof deltaTime === 'number' ? (legacyCC.director.getCurrentTime() + deltaTime * 1000.0) : undefined);
}
```

Reply:
We should use deltaTime instead of now, but that means we need to move out director's calculation to game, so a little more work that this version.
Remove `_lastUpdate`, `_startTime` in director, and implement deltaTime calculation in game

Reply:
To preserve mainLoop API usage, we can add `director.tick`

Reply:
Updated. Please review.